### DISCUSSION TITLE
Printing XLSX using ExcelJS

### DISCUSSION THREAD

**Discussion by souheyltoumi** [exceljs/exceljs#2868] [Q&A]
Hello i am working an XLSX and i need to setup my xlsx so when i print it i get a page with N rows per page (N is a variable i pass) 

imagine i have 144 rows and N = 33:

first page = 33 rows
second page = 33 rows
third page = 33 rows
forth page = 33 rows
last page = 12 rows

i tried everything with rowBreaks but it didn't work any suggestions ?

**Comment by Sarsewar** [comment]
@souheyltoumi <html>
<body>

Try using fitToPage, fitToWidth, fitToHeight Properties. https://github.com/exceljs/exceljs#page-setup

</body>
</html>

**Reply by souheyltoumi** [reply]
thank you but i this solution will depend on which software i use to open my document but the solution i found was to add PageBreaks on wanted rows

**Comment by souheyltoumi** [comment]
```js

    // Add page breaks on rows
    breaks.forEach(break => {
      worksheet.getRow(break).addPageBreak();
    });

```
